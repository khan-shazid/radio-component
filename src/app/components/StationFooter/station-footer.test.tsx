import React from 'react';
import ReactDOM from 'react-dom';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

import Footer from './station-footer';

afterEach(cleanup);

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Footer name='Test'/>, div);
})

it('renders footer correctly', () => {
    const {getByTestId} = render(<Footer name='FM data'/>);
    expect(getByTestId('footer-static')).toHaveTextContent('NOW PLAYING');
    expect(getByTestId('footer-props')).toHaveTextContent('FM data');
})

it('renders footer correctly', () => {
    const {getByTestId} = render(<Footer name=''/>);
    expect(getByTestId('footer-props')).toHaveTextContent('');
})

it('matches snapshot', () => {
    const tree = renderer.create(<Footer name='Fm Data'/>).toJSON();
    expect(tree).toMatchSnapshot();
})
