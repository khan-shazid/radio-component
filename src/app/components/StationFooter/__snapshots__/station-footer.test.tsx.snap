// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="footer"
>
  <label
    data-testid="footer-static"
  >
    NOW PLAYING
  </label>
  <h2
    data-testid="footer-props"
  >
    Fm Data
  </h2>
</div>
`;
